palace/fem/mesh.cpp

```diff
@@ -0,0 +1,369 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "mesh.hpp"
+
+#include "fem/coefficient.hpp"
+#include "fem/fespace.hpp"
+#include "fem/libceed/integrator.hpp"
+
+namespace palace
+{
+
+namespace
+{
+
+const auto &GetParentMesh(const mfem::ParMesh &mesh)
+{
+  // Get the parent mesh if the mesh is a boundary submesh (no submesh of submesh
+  // capabilities, for now).
+  const auto *submesh = dynamic_cast<const mfem::ParSubMesh *>(&mesh);
+  if (submesh && submesh->GetFrom() == mfem::SubMesh::From::Boundary)
+  {
+    return *submesh->GetParent();
+  }
```

Comment:
Not sure I understand why for this. If this is a boundary submesh we want to get the parent (which is the domain mesh), otherwise (domain submesh or just a normal mesh) we can return the mesh object. Are you suggesting we need to check for `submesh && !submesh->GetFrom() == mfem::SubMesh::From::Boundary`?

Reply:
I must've been confused, `submesh && !submesh->GetFrom() == mfem::SubMesh::From::Boundary` is what I was must've been thinking of, but the comment states however that such a mesh we don't care about its parents however. I guess the name threw me, as it's only parents of boundary sub meshes.